test/it/oauth_authenticator_it.js

```diff
@@ -133,6 +133,42 @@ describe('OAuthAuthenticator',function(){
     });
   });
 
+  it('should error when trying to authenticate with invalid id site token', function(done){
+    var authenticator = stormpath.OAuthAuthenticator(application);
+    authenticator.authenticate({
+      body: {
+        grant_type: 'id_site_token',
+        id_site_token: 'abc'
+      }
+    },function(err,result){
+      assert.isUndefined(result);
+
+      assert.isNotNull(err);
+      assert.equal(err.status, 400);
+      assert.equal(err.code, 10017); // 10017 = Token is invalid.
+
+      done();
+    });
+  });
+
+  it('should error when trying to authenticate with invalid refresh token', function(done){
```

Comment:
Yes, this would be a better way to write these tests.  We do a lot of "withs" in it statements, it doesn't feel right.